src/history/hs_rec.c

```diff
@@ -900,6 +900,9 @@ __hs_delete_reinsert_from_pos(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, ui
     WT_UNUSED(key);
 #endif
 
+    /* This function should only accept mixed mode timestamped updates. */
+    WT_ASSERT(session, ts == 1 || ts == WT_TS_NONE);
```

Comment:
Could you add a TODO here it turn this back into a WT_ASSERT_ALWAYS? I don't want to forget to switch it back - we should be able to enforce that.

Reply:
Good point! Updated in [11b5985](https://github.com/wiredtiger/wiredtiger/pull/8268/commits/11b59858ef784730e3719fede5e8a840ecd1469d)